tests/src/Context/MinkContext.php

```diff
@@ -86,4 +89,108 @@ public function iWaitForAjaxToFinish($event = NULL) {
+      if ($selected_remove_buttons = $field->getParent()->findAll('xpath', $xpath)) {
+        // This method selects anew an option in a normal select list. To have
+        // the same results in a select2 widget, all selected options have to
+        // be removed manually.
+        // To select an additional item, use the ::additionallySelectOption
+        // method.
+        foreach ($selected_remove_buttons as $selected_remove_button) {
+          $selected_remove_button->click();
+        }
+      }
+
+      $this->selectSelect2Option($field, $option);
+      return;
+    }
+    parent::selectOption($select, $option);
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * Overrides the parent method in order to support Select2.
+   */
+  public function additionallySelectOption($select, $option): void {
+    if ($field = $this->select2IsUsed($select, $option)) {
+      $this->selectSelect2Option($field, $option);
+      return;
+    }
+    parent::additionallySelectOption($select, $option);
+  }
+
+  /**
+   * Checks if a given select field is using Select2.
+   *
+   * @param string $select
+   *   The select.
+   * @param string $option
+   *   The option to be selected.
+   *
+   * @return \Behat\Mink\Element\NodeElement|false
+   *   It returns the field as node element, if Select2 is used it or FALSE
+   *   otherwise.
+   */
+  protected function select2IsUsed(string $select, string $option) {
+    // In non-Javascript browsers Select2 nicely degrades to a simple select.
+    if (!$this->browserSupportsJavaScript()) {
+      return FALSE;
+    }
+
+    $field = $this->getSession()->getPage()->findField($select);
+    if (!$field->getParent()->find('xpath', '//select[contains(@class, "select2-widget")]')) {
+      // This not a Select2 widget but a simple select.
+      return FALSE;
+    }
+    return $field;
+  }
+
+  /**
+   * Selects a Select2 option.
+   *
+   * @param \Behat\Mink\Element\NodeElement $field
+   *   The select field as a node element.
+   * @param string $option
+   *   The option to be selected.
+   */
+  protected function selectSelect2Option(NodeElement $field, string $option): void {
+    $select2_search = $field->getParent()->find('xpath', '//select[contains(@class, "select2-widget")]/following-sibling::span//input[@class="select2-search__field"]');
+    $select2_search->setValue($option);
+    $select2_option = $this->getSession()->getPage()->find('xpath', '//li[contains(text(), "' . $option . '")]');
+    $select2_option->click();
+  }
+
+  /**
+   * Deselects a Select2 option by clicking the x button in the chip.
+   *
+   * @param string $option
+   *   The option to be deselected.
+   * @param string $select
+   *   The select field name.
+   *
+   * @throws \Exception
+   *   Thrown if the field is not a select2 widget.
+   *
+   * @Given I deselect the option :option from the :select select2 widget
```

Comment:
I cannot find any usage of this step definition but let's keep it, it might be useful in the future.

Again s/select2/Select2 :)